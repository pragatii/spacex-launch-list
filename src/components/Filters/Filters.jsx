import {Box, Divider, Grid, Typography} from "@material-ui/core";
import * as React from 'react';
import Chip from "@material-ui/core/Chip";

export default function Filters(props) {
    const {title, options} = props;

    return (
        <Box display={'flex'} flexDirection={'column'}>
            <Typography component={"body"}>
                {title}
            </Typography>
            <Divider/>
            <Grid container spacing={2}>
                {options.map(option => <Grid item>
                    <Chip label={option}/>
                </Grid>)}
            </Grid>
        </Box>
    );
}
