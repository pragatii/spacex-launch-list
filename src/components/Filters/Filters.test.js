import React from 'react';
import { mount, shallow, render } from 'enzyme';

import Filters from './Filters';

const props = {
    title: '',
    options: [1, 2, 3],
    classes: {},
    onSelect: jest.fn(),
    selectedFilter: ''
}
it('renders correctly', () => {
    let wrapper = shallow(<Filters/>);
    expect(wrapper).toMatchSnapshot();
});

it('should mount the component with all props', () => {
    const component = mount(<Filters {...props} />);
    expect(component).toMatchSnapshot();
    component.unmount();
});
