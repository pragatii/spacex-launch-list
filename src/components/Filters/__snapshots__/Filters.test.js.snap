// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `ShallowWrapper {}`;

exports[`should be possible to activate button with Spacebar 1`] = `ReactWrapper {}`;

exports[`should mount the component with all props 1`] = `ReactWrapper {}`;
